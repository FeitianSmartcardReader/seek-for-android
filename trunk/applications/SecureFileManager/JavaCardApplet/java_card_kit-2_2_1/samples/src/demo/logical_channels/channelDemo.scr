
//+
// Copyright © 2003 Sun Microsystems, Inc. All rights reserved.
// SUN PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
//-

//+
// Workfile:@(#)Header.scr	1.8
// Version:1.8
// Date:07/16/03
//-

powerup;

// Select the installer applet
0x00 0xA4 0x04 0x00 0x09 0xa0 0x00 0x00 0x00 0x62 0x03 0x01 0x08 0x01 0x7F;
// 90 00 = SW_NO_ERROR

// Turn output off during CAP file download
echo "CAP file download section. Output suppressed.";

output off;

0x80 0xB0 0x00 0x00 0x00 0x7F;

// com/sun/javacard/samples/ChannelsDemo/javacard/Header.cap
0x80 0xB2 0x01 0x00 0x00 0x7F;
0x80 0xB4 0x01 0x00 0x16 0x01 0x00 0x13 0xDE 0xCA 0xFF 0xED 0x01 0x02 0x04 0x00 0x01 0x09 0xA0 0x00 0x00 0x00 0x62 0x03 0x01 0x0C 0x0B 0x7F;
0x80 0xBC 0x01 0x00 0x00 0x7F;

// com/sun/javacard/samples/ChannelsDemo/javacard/Directory.cap
0x80 0xB2 0x02 0x00 0x00 0x7F;
0x80 0xB4 0x02 0x00 0x20 0x02 0x00 0x1F 0x00 0x13 0x00 0x1F 0x00 0x1B 0x00 0x0B 0x00 0x8E 0x00 0x38 0x03 0x22 0x00 0x17 0x00 0x63 0x00 0x00 0x01 0xD3 0x00 0x04 0x00 0x01 0x00 0x0A 0x01 0x7F;
0x80 0xB4 0x02 0x00 0x02 0x02 0x00 0x7F;
0x80 0xBC 0x02 0x00 0x00 0x7F;

// com/sun/javacard/samples/ChannelsDemo/javacard/Import.cap
0x80 0xB2 0x04 0x00 0x00 0x7F;
0x80 0xB4 0x04 0x00 0x0E 0x04 0x00 0x0B 0x01 0x02 0x01 0x07 0xA0 0x00 0x00 0x00 0x62 0x01 0x01 0x7F;
0x80 0xBC 0x04 0x00 0x00 0x7F;

// com/sun/javacard/samples/ChannelsDemo/javacard/Applet.cap
0x80 0xB2 0x03 0x00 0x00 0x7F;
0x80 0xB4 0x03 0x00 0x1E 0x03 0x00 0x1B 0x02 0x0A 0xA0 0x00 0x00 0x00 0x62 0x03 0x01 0x0C 0x0B 0x01 0x00 0x61 0x0A 0xA0 0x00 0x00 0x00 0x62 0x03 0x01 0x0C 0x0B 0x02 0x01 0xC7 0x7F;
0x80 0xBC 0x03 0x00 0x00 0x7F;

// com/sun/javacard/samples/ChannelsDemo/javacard/Class.cap
0x80 0xB2 0x06 0x00 0x00 0x7F;
0x80 0xB4 0x06 0x00 0x20 0x06 0x00 0x38 0x01 0x80 0x03 0x03 0x00 0x01 0x04 0x06 0x00 0x01 0x00 0x73 0xFF 0xFF 0x00 0x6F 0x00 0x7D 0x00 0x76 0x00 0x7A 0x00 0xCA 0x80 0x11 0x02 0x08 0x09 0x7F;
0x80 0xB4 0x06 0x00 0x1B 0x01 0x80 0x03 0x02 0x00 0x02 0x04 0x06 0x00 0x00 0x01 0xDD 0xFF 0xFF 0x01 0xD5 0x01 0xF8 0x01 0xE4 0x01 0xF1 0x80 0x11 0x02 0x08 0x09 0x7F;
0x80 0xBC 0x06 0x00 0x00 0x7F;

// com/sun/javacard/samples/ChannelsDemo/javacard/Method.cap
0x80 0xB2 0x07 0x00 0x00 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x07 0x03 0x22 0x00 0x07 0x40 0x18 0x8C 0x00 0x16 0x18 0x05 0x90 0x0C 0x87 0x00 0x18 0x7F 0x00 0x18 0x18 0x03 0x89 0x01 0x18 0x19 0x1E 0x59 0x02 0x01 0x25 0x10 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x08 0x4D 0x19 0x1E 0x59 0x02 0x01 0x25 0x11 0x00 0xFF 0x53 0x55 0x89 0x02 0xAD 0x00 0x03 0x19 0x1E 0x59 0x02 0x01 0x25 0x10 0x08 0x4D 0x19 0x1E 0x59 0x02 0x01 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x25 0x11 0x00 0xFF 0x53 0x55 0x39 0xAD 0x00 0x04 0x19 0x1E 0x59 0x02 0x01 0x25 0x10 0x08 0x4D 0x19 0x1E 0x59 0x02 0x01 0x25 0x11 0x00 0xFF 0x53 0x55 0x39 0x18 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x8B 0x00 0x05 0x7A 0x05 0x30 0x8F 0x00 0x06 0x3D 0x18 0x1D 0x1E 0x8C 0x00 0x07 0x3B 0x7A 0x01 0x10 0x04 0x78 0x00 0x10 0x7A 0x01 0x20 0x04 0x78 0x00 0x20 0x7A 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x04 0x21 0x19 0x8B 0x00 0x08 0x2D 0x1A 0x03 0x1A 0x03 0x25 0x10 0xFC 0x53 0x38 0x1A 0x03 0x25 0x61 0x0A 0x1A 0x04 0x25 0x10 0xA4 0x6B 0x03 0x7A 0x1A 0x03 0x25 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x10 0x80 0x6A 0x08 0x11 0x6E 0x00 0x8D 0x00 0x09 0x1A 0x04 0x25 0x75 0x00 0x19 0x00 0x02 0x00 0x10 0x00 0x0D 0x00 0x20 0x00 0x13 0x18 0x19 0x8C 0x00 0x0A 0x7A 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x18 0x19 0x8C 0x00 0x0B 0x7A 0x11 0x6D 0x00 0x8D 0x00 0x09 0x7A 0x03 0x21 0x1D 0xAF 0x02 0x6B 0x09 0xAD 0x00 0x03 0x26 0x31 0x70 0x07 0xAD 0x00 0x04 0x26 0x31 0x7F;
0x80 0xB4 0x07 0x00 0x20 0xAF 0x01 0x1E 0x6C 0x11 0x8D 0x00 0x0C 0x18 0xAF 0x01 0x1E 0x43 0x89 0x01 0x8D 0x00 0x0D 0x04 0x78 0x03 0x78 0x01 0x00 0x7B 0x00 0x18 0x77 0x03 0x24 0x19 0x8B 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x00 0x08 0x2D 0x1A 0x07 0x25 0x32 0x19 0x8B 0x00 0x0E 0x5B 0x29 0x04 0x1F 0x05 0x6B 0x07 0x16 0x04 0x05 0x6A 0x08 0x11 0x67 0x00 0x8D 0x00 0x09 0x1A 0x08 0x25 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x10 0x08 0x4D 0x1A 0x10 0x06 0x25 0x55 0x29 0x05 0x16 0x05 0x11 0x7F 0xFF 0x6E 0x06 0x16 0x05 0x63 0x08 0x11 0x6A 0x55 0x8D 0x00 0x09 0xAF 0x01 0x16 0x05 0x41 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x11 0x7F 0xFF 0x6F 0x08 0x11 0x6A 0x54 0x8D 0x00 0x09 0x8D 0x00 0x0C 0x18 0xAF 0x01 0x16 0x05 0x41 0x89 0x01 0x8D 0x00 0x0D 0x7A 0x04 0x23 0x7B 0x00 0x0F 0x03 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x7B 0x00 0x0F 0x92 0x5B 0x8D 0x00 0x10 0x2D 0x1A 0x8D 0x00 0x11 0x61 0x08 0x11 0x69 0x05 0x8D 0x00 0x09 0x19 0x8B 0x00 0x08 0x2E 0x19 0x8B 0x00 0x12 0x29 0x04 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x16 0x04 0x05 0x6D 0x08 0x11 0x67 0x00 0x8D 0x00 0x09 0x19 0x05 0x8B 0x00 0x13 0x1B 0x03 0xAF 0x01 0x10 0x08 0x4F 0x5B 0x38 0x1B 0x04 0xAF 0x01 0x11 0x00 0xFF 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x53 0x5B 0x38 0x19 0x03 0x05 0x8B 0x00 0x14 0x7A 0x03 0x40 0x18 0x8C 0x00 0x16 0x18 0x04 0x05 0x8D 0x00 0x15 0x87 0x03 0xAD 0x03 0x03 0x02 0x39 0x18 0x04 0x05 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x8D 0x00 0x17 0x87 0x04 0x18 0x8B 0x00 0x19 0x7A 0x05 0x30 0x8F 0x00 0x1A 0x3D 0x18 0x1D 0x1E 0x8C 0x00 0x1B 0x3B 0x7A 0x01 0x10 0x18 0x8C 0x00 0x1C 0x04 0x78 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x01 0x10 0x18 0x8C 0x00 0x1D 0x7A 0x01 0x20 0x1D 0x60 0x04 0x03 0x78 0x18 0x8C 0x00 0x1C 0x04 0x78 0x01 0x20 0x18 0x8C 0x00 0x1D 0x7A 0x04 0x21 0x19 0x8B 0x00 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x08 0x2D 0x1A 0x03 0x1A 0x03 0x25 0x10 0xFC 0x53 0x38 0x1A 0x03 0x25 0x61 0x0A 0x1A 0x04 0x25 0x10 0xA4 0x6B 0x03 0x7A 0x1A 0x03 0x25 0x10 0x80 0x6A 0x08 0x11 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x6E 0x00 0x8D 0x00 0x09 0x1A 0x04 0x25 0x75 0x00 0x21 0x00 0x03 0x00 0x10 0x00 0x11 0x00 0x20 0x00 0x17 0x00 0x30 0x00 0x1C 0x18 0x19 0x8C 0x00 0x1E 0x7A 0x18 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x8C 0x00 0x1F 0x7A 0x18 0x8C 0x00 0x20 0x7A 0x11 0x6D 0x00 0x8D 0x00 0x09 0x7A 0x03 0x24 0x19 0x8B 0x00 0x08 0x2D 0x1A 0x07 0x25 0x32 0x19 0x8B 0x00 0x0E 0x5B 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x29 0x04 0x1F 0x05 0x6B 0x07 0x16 0x04 0x05 0x6A 0x08 0x11 0x67 0x00 0x8D 0x00 0x09 0x1A 0x08 0x25 0x10 0x08 0x4D 0x1A 0x10 0x06 0x25 0x11 0x00 0xFF 0x53 0x55 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x29 0x05 0x16 0x05 0x02 0x6A 0x0A 0xAD 0x03 0x03 0x16 0x05 0x39 0x70 0x0C 0x18 0x8C 0x00 0x20 0x11 0x6A 0x52 0x8D 0x00 0x09 0xAD 0x04 0x03 0x25 0x04 0x6B 0x23 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x8D 0x00 0x21 0x67 0x08 0x11 0x6A 0x53 0x8D 0x00 0x09 0x8D 0x00 0x21 0xAD 0x03 0x03 0x26 0x8B 0x00 0x22 0x61 0x0C 0x18 0x8C 0x00 0x20 0x11 0x6A 0x51 0x8D 0x00 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x09 0x7A 0x03 0x10 0x8D 0x00 0x21 0x67 0x08 0x11 0x6A 0x53 0x8D 0x00 0x09 0xAD 0x04 0x03 0x25 0x04 0x6B 0x08 0x11 0x6A 0x50 0x8D 0x00 0x09 0xAD 0x03 0x03 0x26 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x02 0x6B 0x08 0x11 0x6A 0x52 0x8D 0x00 0x09 0x8D 0x00 0x21 0xAD 0x03 0x03 0x26 0x8B 0x00 0x22 0x60 0x09 0xAD 0x04 0x03 0x04 0x38 0x70 0x08 0x11 0x6A 0x51 0x8D 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x00 0x09 0x7A 0x03 0x10 0xAD 0x04 0x03 0x03 0x38 0x7A 0x03 0x10 0xAD 0x04 0x03 0x03 0x38 0xAD 0x03 0x03 0x02 0x39 0x7A 0x03 0x10 0xAD 0x03 0x03 0x02 0x39 0xAD 0x7F;
0x80 0xB4 0x07 0x00 0x05 0x04 0x03 0x03 0x38 0x7A 0x7F;
0x80 0xBC 0x07 0x00 0x00 0x7F;

// com/sun/javacard/samples/ChannelsDemo/javacard/StaticField.cap
0x80 0xB2 0x08 0x00 0x00 0x7F;
0x80 0xB4 0x08 0x00 0x1A 0x08 0x00 0x17 0x00 0x04 0x00 0x02 0x00 0x01 0x03 0x00 0x0A 0xA0 0x00 0x00 0x00 0x62 0x03 0x01 0x0C 0x0B 0x02 0x00 0x00 0x00 0x00 0x7F;
0x80 0xBC 0x08 0x00 0x00 0x7F;

// com/sun/javacard/samples/ChannelsDemo/javacard/ConstantPool.cap
0x80 0xB2 0x05 0x00 0x00 0x7F;
0x80 0xB4 0x05 0x00 0x20 0x05 0x00 0x8E 0x00 0x23 0x02 0x00 0x00 0x00 0x02 0x00 0x00 0x01 0x02 0x00 0x00 0x02 0x02 0x00 0x1D 0x00 0x02 0x00 0x1D 0x01 0x03 0x80 0x03 0x01 0x01 0x00 0x00 0x7F;
0x80 0xB4 0x05 0x00 0x20 0x00 0x06 0x00 0x00 0x01 0x03 0x80 0x0A 0x01 0x06 0x80 0x07 0x01 0x06 0x00 0x01 0x57 0x06 0x00 0x00 0xF9 0x06 0x80 0x08 0x01 0x06 0x80 0x08 0x02 0x03 0x80 0x0A 0x7F;
0x80 0xB4 0x05 0x00 0x20 0x06 0x05 0x00 0x00 0x00 0x06 0x80 0x08 0x0B 0x06 0x80 0x08 0x14 0x03 0x80 0x0A 0x07 0x03 0x80 0x0A 0x09 0x03 0x80 0x0A 0x04 0x06 0x80 0x08 0x0F 0x06 0x80 0x03 0x7F;
0x80 0xB4 0x05 0x00 0x20 0x00 0x06 0x80 0x08 0x0D 0x05 0x00 0x00 0x02 0x03 0x80 0x03 0x01 0x01 0x00 0x1D 0x00 0x06 0x00 0x01 0xA7 0x06 0x00 0x03 0x08 0x06 0x00 0x03 0x15 0x06 0x00 0x02 0x7F;
0x80 0xB4 0x05 0x00 0x11 0x4D 0x06 0x00 0x02 0xBF 0x06 0x00 0x03 0x00 0x06 0x00 0x00 0xF3 0x03 0x00 0x00 0x80 0x7F;
0x80 0xBC 0x05 0x00 0x00 0x7F;

// com/sun/javacard/samples/ChannelsDemo/javacard/RefLocation.cap
0x80 0xB2 0x09 0x00 0x00 0x7F;
0x80 0xB4 0x09 0x00 0x20 0x09 0x00 0x63 0x00 0x1F 0x0C 0x08 0x17 0x02 0x18 0x89 0x04 0x07 0x05 0x09 0x04 0x4E 0x14 0x05 0x3E 0x09 0x1B 0x02 0x0B 0xC4 0x12 0x15 0x21 0x0D 0x10 0x09 0x10 0x7F;
0x80 0xB4 0x09 0x00 0x20 0x08 0x05 0x08 0x05 0x00 0x40 0x05 0x0A 0x4F 0x06 0x07 0x16 0x24 0x15 0x06 0x07 0x1C 0x0A 0x09 0x07 0x09 0x12 0x1E 0x10 0x03 0x0B 0x06 0x04 0x05 0x05 0x08 0x04 0x7F;
0x80 0xB4 0x09 0x00 0x20 0x05 0x0D 0x05 0x19 0x07 0x06 0x0D 0x06 0x06 0x07 0x08 0x08 0x0C 0x08 0x07 0x24 0x19 0x05 0x05 0x07 0x07 0x09 0x12 0x22 0x06 0x0A 0x08 0x03 0x07 0x06 0x06 0x06 0x7F;
0x80 0xB4 0x09 0x00 0x06 0x08 0x0D 0x0D 0x03 0x07 0x0F 0x7F;
0x80 0xBC 0x09 0x00 0x00 0x7F;

0x80 0xBA 0x00 0x00 0x00 0x7F;
//+
// Copyright © 2003 Sun Microsystems, Inc. All rights reserved.
// SUN PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
//-

//+
// Workfile:@(#)channel.scr	1.5
// Version:1.5
// Date:07/16/03
//-

// Logical Channels demo

// This demo simulates a ficticious wireless device using a smart card 
// to keep track of use time in a fictitious network.
// The device user only has a certain amount of credits he can use, and
// every unit of time he remains connected to the network using the
// device decreases such credits.
// The user is assigned a home area, where he can operate the device at
// a lower cost.  However, if he "moves" to an external area, the 
// network connection charges increase.  If the user runs out of credits,
// the user is disconnected from the network autmatically.

// resume printing output during functional test
output on;

echo "Applet creation and functional test section.";

// Select the installer applet
0x00 0xA4 0x04 0x00 0x09 0xa0 0x00 0x00 0x00 0x62 0x03 0x01 0x08 0x01 0x7F;
// 90 00 = SW_NO_ERROR

// create ConnectionManager
0x80 0xB8 0x00 0x00 0x0c 0x0a 0xa0 0x00 0x00 0x00 0x62 0x03 0x01 0x0c 0x0b 0x02 0x00 0x7F;

// create AccountAccessor
0x80 0xB8 0x00 0x00 0x12 0x0a 0xa0 0x00 0x00 0x00 0x62 0x03 0x01 0x0c 0x0b 0x01 0x08 0x01 0x98 0x00 0x01 0x00 0x05 0x7F;

// Select Account accessor
0x00 0xa4 0x04 0x00 0x0a 0xa0 0x00 0x00 0x00 0x62 0x03 0x01 0x0c 0x0b 0x01 0x7F;
// 90 00 = SW_NO_ERROR

// Initialize Account accessor with 100 credits
0x80 0x20 0x00 0x00 0x02 0x00 0x64 0x7F;
// 90 00 = SW_NO_ERROR

// Assume some time happens, and the user turns the device on to use it...

// Terminal selects Connection Manager upon turing on the device
0x00 0xa4 0x04 0x00 0x0a 0xa0 0x00 0x00 0x00 0x62 0x03 0x01 0x0c 0x0b 0x02 0x7F;
// 90 00 = SW_NO_ERROR

// Upon network carrier detect, terminal decodes area code (408 dec) 
// and sends a "Time tick" to the card
0x80 0x10 0x00 0x00 0x02 0x01 0x98 0x7F;
// 90 00 = SW_NO_ERROR


// The user starts using the device.  The terminal quieries for available credits.
// First, open a new logical channel, via MANAGE CHANNEL OPENcommand.
0x00 0x70 0x00 0x00 0x00 0x01;

// In order to do so without losing network information (in ConnectionManager 
// applet), Account Accessor applet is selected on a second logical channel.
0x01 0xa4 0x04 0x00 0x0a 0xa0 0x00 0x00 0x00 0x62 0x03 0x01 0x0c 0x0b 0x01 0x7F;
// 90 00 = SW_NO_ERROR

// Terminal queries balance on user account.
0x81 0x10 0x00 0x00 0x00 0x02;
// 0x00 0x64 0x90 0x00 = 100 credits and SW_NO_ERROR

// User starts a network service.  For this, sufficient credits shall be available
// on the user account to pay for the first time unot of device use.
0x80 0x20 0x00 0x00 0x00 0x7F;
// 90 00 = SW_NO_ERROR

// Service initiation has a charge of one time unit.  Verified by obtaining balance.
0x81 0x10 0x00 0x00 0x00 0x02;
// 0x00 0x63 0x90 0x00 = SW_NO_ERROR

// Time tick for another time unit
0x80 0x10 0x00 0x00 0x02 0x01 0x98 0x7F;
// 90 00 = SW_NO_ERROR

// Terminal quieries balance. 
0x81 0x10 0x00 0x00 0x00 0x02;
// 0x00 0x62 0x90 0x00 = SW_NO_ERROR

// Time tick.  The user has moved to a new area code (650).
0x80 0x10 0x00 0x00 0x02 0x02 0x8A 0x7F;
// 90 00 = SW_NO_ERROR

// Terminal quieries balance. 
0x81 0x10 0x00 0x00 0x00 0x02;
// 0x00 0x5D 0x90 0x00 = SW_NO_ERROR

// Time tick.  Area code (650).
0x80 0x10 0x00 0x00 0x02 0x02 0x8A 0x7F;
// 90 00 = SW_NO_ERROR

// Terminal quieries balance. 
0x81 0x10 0x00 0x00 0x00 0x02;
// 0x00 0x58 0x90 0x00 = SW_NO_ERROR

// User terminates network service. No charges applied.
0x80 0x30 0x00 0x00 0x00 0x7F;
// 90 00 = SW_NO_ERROR

// Time tick.  Area code (650).
0x80 0x10 0x00 0x00 0x02 0x02 0x8A 0x7F;
// 90 00 = SW_NO_ERROR

// Terminal quieries balance. No change, since device is not in use.
0x81 0x10 0x00 0x00 0x00 0x02;
// 0x00 0x58 0x90 0x00 = SW_NO_ERROR

// The Account manager applet is deselected,
// via a MANAGE CHANNEL CLOSE command.
0x00 0x70 0x80 0x01 0x00 0x7F;

// User powers down the device. 

powerdown;

