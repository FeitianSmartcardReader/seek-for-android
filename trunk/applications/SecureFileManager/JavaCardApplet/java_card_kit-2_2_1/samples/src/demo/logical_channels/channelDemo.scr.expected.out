CLA: 00, INS: a4, P1: 04, P2: 00, Lc: 09, a0, 00, 00, 00, 62, 03, 01, 08, 01, Le: 00, SW1: 90, SW2: 00
CAP file download section. Output suppressed.
OUTPUT OFF;
OUTPUT ON;
Applet creation and functional test section.
CLA: 00, INS: a4, P1: 04, P2: 00, Lc: 09, a0, 00, 00, 00, 62, 03, 01, 08, 01, Le: 00, SW1: 90, SW2: 00
CLA: 80, INS: b8, P1: 00, P2: 00, Lc: 0c, 0a, a0, 00, 00, 00, 62, 03, 01, 0c, 0b, 02, 00, Le: 0a, a0, 00, 00, 00, 62, 03, 01, 0c, 0b, 02, SW1: 90, SW2: 00
CLA: 80, INS: b8, P1: 00, P2: 00, Lc: 12, 0a, a0, 00, 00, 00, 62, 03, 01, 0c, 0b, 01, 08, 01, 98, 00, 01, 00, 05, Le: 0a, a0, 00, 00, 00, 62, 03, 01, 0c, 0b, 01, SW1: 90, SW2: 00
CLA: 00, INS: a4, P1: 04, P2: 00, Lc: 0a, a0, 00, 00, 00, 62, 03, 01, 0c, 0b, 01, Le: 00, SW1: 90, SW2: 00
CLA: 80, INS: 20, P1: 00, P2: 00, Lc: 02, 00, 64, Le: 00, SW1: 90, SW2: 00
CLA: 00, INS: a4, P1: 04, P2: 00, Lc: 0a, a0, 00, 00, 00, 62, 03, 01, 0c, 0b, 02, Le: 00, SW1: 90, SW2: 00
CLA: 80, INS: 10, P1: 00, P2: 00, Lc: 02, 01, 98, Le: 00, SW1: 90, SW2: 00
CLA: 00, INS: 70, P1: 00, P2: 00, Lc: 00, Le: 01, 01, SW1: 90, SW2: 00
CLA: 01, INS: a4, P1: 04, P2: 00, Lc: 0a, a0, 00, 00, 00, 62, 03, 01, 0c, 0b, 01, Le: 00, SW1: 90, SW2: 00
CLA: 81, INS: 10, P1: 00, P2: 00, Lc: 00, Le: 02, 00, 64, SW1: 90, SW2: 00
CLA: 80, INS: 20, P1: 00, P2: 00, Lc: 00, Le: 00, SW1: 90, SW2: 00
CLA: 81, INS: 10, P1: 00, P2: 00, Lc: 00, Le: 02, 00, 63, SW1: 90, SW2: 00
CLA: 80, INS: 10, P1: 00, P2: 00, Lc: 02, 01, 98, Le: 00, SW1: 90, SW2: 00
CLA: 81, INS: 10, P1: 00, P2: 00, Lc: 00, Le: 02, 00, 62, SW1: 90, SW2: 00
CLA: 80, INS: 10, P1: 00, P2: 00, Lc: 02, 02, 8a, Le: 00, SW1: 90, SW2: 00
CLA: 81, INS: 10, P1: 00, P2: 00, Lc: 00, Le: 02, 00, 5d, SW1: 90, SW2: 00
CLA: 80, INS: 10, P1: 00, P2: 00, Lc: 02, 02, 8a, Le: 00, SW1: 90, SW2: 00
CLA: 81, INS: 10, P1: 00, P2: 00, Lc: 00, Le: 02, 00, 58, SW1: 90, SW2: 00
CLA: 80, INS: 30, P1: 00, P2: 00, Lc: 00, Le: 00, SW1: 90, SW2: 00
CLA: 80, INS: 10, P1: 00, P2: 00, Lc: 02, 02, 8a, Le: 00, SW1: 90, SW2: 00
CLA: 81, INS: 10, P1: 00, P2: 00, Lc: 00, Le: 02, 00, 58, SW1: 90, SW2: 00
CLA: 00, INS: 70, P1: 80, P2: 01, Lc: 00, Le: 00, SW1: 90, SW2: 00
